src/Mvc/test/Microsoft.AspNetCore.Mvc.Core.Test/Routing/UrlHelperExtensionsTest.cs

```diff
@@ -601,6 +602,112 @@ public void Page_UsesAreaValueFromRouteValueIfSpecified()
+            var actionContext = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+            var urlHelper = CreateMockUrlHelper(actionContext);
+            urlHelper.Setup(h => h.Action(It.IsAny<UrlActionContext>()))
+                .Callback((UrlActionContext context) => actual = context);
+
+            // Act
+            urlHelper.Object.ActionLink(expectedAction);
+
+            // Assert
+            urlHelper.Verify();
+            Assert.NotNull(actual);
+            Assert.Equal(expectedAction, actual.Action);
+            Assert.Null(actual.Controller);
+            Assert.Null(actual.Values);
+
+            Assert.Equal(expectedProtocol, actual.Protocol);
+            Assert.Equal(expectedHost, actual.Host);
+        }
+
+        [Fact]
+        public void ActionLink_UsesSpecifiedProtocol()
+        {
+            // Arrange
+            var expectedProtocol = "testprotocol://";
+            var expectedHost = "www.example.com";
+            UrlActionContext actual = null;
+
+            var httpContext = new DefaultHttpContext
+            {
+                Request =
+                {
+                    Protocol = "http://",
+                    Host = new HostString(expectedHost),
+                }
+            };
+            var actionContext = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+            var urlHelper = CreateMockUrlHelper(actionContext);
+            urlHelper.Setup(h => h.Action(It.IsAny<UrlActionContext>()))
+                .Callback((UrlActionContext context) => actual = context);
+
+            // Act
+            urlHelper.Object.ActionLink(protocol: expectedProtocol);
+
+            // Assert
+            urlHelper.Verify();
+            Assert.NotNull(actual);
+            Assert.Null(actual.Action);
+            Assert.Null(actual.Controller);
+            Assert.Null(actual.Values);
+
+            Assert.Equal(expectedProtocol, actual.Protocol);
+            Assert.Equal(expectedHost, actual.Host);
+        }
+
+        [Fact]
+        public void ActionLink_UsesSpecifiedHost()
+        {
+            // Arrange
+            var expectedProtocol = "testprotocol://";
+            var expectedHost = "www.example.com";
+            UrlActionContext actual = null;
+
+            var httpContext = new DefaultHttpContext
+            {
+                Request =
+                {
+                    Protocol = "http://",
+                    Host = new HostString("www.asp.net"),
+                }
+            };
+            var actionContext = new ActionContext(httpContext, new RouteData(), new ActionDescriptor());
+            var urlHelper = CreateMockUrlHelper(actionContext);
+            urlHelper.Setup(h => h.Action(It.IsAny<UrlActionContext>()))
+                .Callback((UrlActionContext context) => actual = context);
+
+            // Act
+            urlHelper.Object.ActionLink(protocol: expectedProtocol, host: expectedHost);
```

Comment:
Suggest calling this w/ just expected host and using `expectedProtocol` when creating the context. That would isolate the tests a bit more.